kong/plugins/response-ratelimiting/policies/init.lua

```diff
@@ -85,9 +85,13 @@ local function get_redis_connection(conf)
     if is_present(conf.redis_password) then
       local ok, err
       if is_present(conf.redis_username) then
-        ok, err = red:auth(conf.redis_username, conf.redis_password)
+        ok, err = kong.vault.try(function(cfg)
+          return red:auth(cfg.redis_username, cfg.redis_password)
+        end, conf)
```

Comment:
Aapo, was the plan not to keep this transparent to the plugin authors and resolve the referenced username/password before calling any plugin handler code?

I'm surprised to see this code here.

Reply:
@hbagdi, this is the `resolve-secret-on-failure` approach. Most of the cases we have with secrets are `resolve-on-secret-failure`. But we decided to focus on more generic `ttl`, but that has limitations. E.g. with `try`, you can just go to vault and change secret and then change db password, and Kong should be able to pick it up. The TTL approach requires a more coordinated approach. That is when rotating, you should keep the old one at least for TTL amount of time, and after that you may drop the old credentials. The TTL approach can be seen in action here (the update function): https://github.com/Kong/kong/pull/10573/commits/085833ce4633e282a276db21a44f704c7cd6cd6f

But the TTL we currently only re-resolve secrets only every minute as most:
https://github.com/Kong/kong/pull/10573/commits/20c2da810c4b40995d7cb2c3fcbf54620e49099f#diff-212b3488c5eb070820f7bc5178fbaa56e9435fe49cf050feb29091ca9aa1540bR49